temp/legacy-code/src/main/java/com/ivy/legacy/legacy/ui/theme/modal/edit/CalculatorModal.kt

```diff
@@ -178,6 +178,17 @@ fun BoxWithConstraintsScope.CalculatorModal(
         Spacer(Modifier.height(24.dp))
     }
 }
+private fun handleOperator(expression: String, operator: String): String {
+    return if (expression.isNotEmpty() && expression.last().isOperator()) {
+        expression.dropLast(1) + operator
+    } else {
+        expression + operator
+    }
+}
+
+private fun Char.isOperator(): Boolean {
+    return this == '+' || this == '−' || this == '×' || this == '÷'
```

Comment:
Use 
```
fun Char.isOperator(): Boolean = when(this) {
  '+',  '-' -> true 
  else -> false
}
```

Reply:
> ```kotlin
> fun Char.isOperator(): Boolean = when(this) {
>   '+', '-' -> true 
>   else -> false
> }
> ```

if we use this then the ' x ' and ' ÷ ' will get added consecutively

Reply:
> yeah for all operators was lazy to write, just use `when` so it's more idiomatic and simple

      fun Char.isOperator(): Boolean = when(this) {
    '+', '-','×', '÷' -> true
    else -> false
      }

is this ok ?

Reply:
yup